File: Lab3_19/4.py
'''
Численно найти сумму функционального ряда (k=1, infty)(-1)**k / (sin(k * x) + 19 + k**2) с точностью 10^-6
'''
from math import *

def calculate_term(k, x):
    return (-1)**k / (sin(k * x) + 19 + k**2) #Функциональный ряд

# Функция для вычисления суммы ряда с заданной точностью
def calculate_sum(x, tolerance):
    k = 1
    sum_result = 0.0
    term = calculate_term(k, x)

    while abs(term) >= tolerance:
        sum_result += term
        k += 1
        term = calculate_term(k, x)

    return sum_result

tolerance = 1e-6  # Устанавливаем требуемую точность
x = 5  # Задаем значение x

result = calculate_sum(x, tolerance)  # Вычисляем сумму функционального ряда

# Выводим результат с заданной точностью
print("Ряд (k=1, infty)(-1)**k / (sin(k * x) + 19 + k**2)")
print(f"Сумма функционального ряда с точностью 10^-6 при x = {x}:")
print(f"Sum = {result}")

File: Lab3_19/4.cpp
/*
Численно найти сумму функционального ряда (k=1, infty)(-1)**k / (sin(k * x) + 19 + k**2) с точностью 10^-6
*/
#include <iostream>
#include <cmath>
#include <iomanip>

using namespace std;

// Функциональный ряд
double calculateTerm(int k, double x) {
    return pow(-1, k) / (sin(k * x) + 19 + pow(k, 2));
}

// Функция для вычисления суммы ряда с заданной точностью
double calculateSum(double x, double tolerance) {
    double sum = 0.0;
    double term = 0.0;
    int k = 1;

    do {
        term = calculateTerm(k, x);// Вычисляем очередной член ряда
        sum += term;//приибавляем 
        k++; // к следующему
    } while (std::abs(term) >= tolerance);//остановка при достижении точности

    return sum;
}

int main() {
    double tolerance = 1e-6; // Устанавливаем требуемую точность
    double x = 5.0; // Устанавливаем значение x
    double result = calculateSum(x, tolerance); // Вычисляем сумму функционального ряда

    cout << "Ряд (k=1, infty)(-1)**k / (sin(k * x) + 19 + k**2)"<< endl;
     cout << "Сумма функционального ряда с точностью 10^-6 при x = " << x << ":" << std::endl;
     cout << "Sum = "<< fixed << setprecision(12) << result << endl;
    
    return 0;
}